java/daikon/chicory/Instrument24.java

```diff
@@ -0,0 +1,1720 @@
+      t.printStackTrace();
+      // No changes to the bytecodes
+      return null;
+    }
+
+    if (Chicory.dump) {
+      try {
+        debug_transform.log(
+            "Dumping .class and .javap for %s to %s%n", binaryClassName, debug_orig_dir);
+        // Define the output file
+        Path outputFile = Paths.get(debug_orig_dir.toString(), binaryClassName + ".class");
+        // Write the byte array to a .class file.
+        Files.write(outputFile, classFile.transformClass(classModel, ClassTransform.ACCEPT_ALL));
+        // Write a BCEL-like file with an extension of .javap.
+
+        try (BufferedWriter writer =
+            Files.newBufferedWriter(
+                new File(debug_orig_dir, binaryClassName + ".javap").toPath(),
+                StandardCharsets.UTF_8)) {
+          writer.write(classModel.toDebugString());
+        }
+      } catch (Throwable t) {
+        System.err.printf(
+            "Unexpected error %s dumping out debug files for: %s%n", t, binaryClassName);
+        t.printStackTrace();
+        // proceed with instrumentation
+      }
+    }
+
+    hasClinit = false;
+    byte[] newBytes = {};
+    debug_transform.log("%nClass: %s%n", binaryClassName);
+    // Instrument the classfile, die on any errors
+    try {
+      newBytes =
+          classFile.build(
+              classModel.thisClass().asSymbol(),
+              classBuilder -> instrumentClass(classBuilder, classModel, cfLoader));
+
+    } catch (Throwable t) {
+      System.err.printf("Unexpected error %s in transform of %s%n", t, binaryClassName);
+      t.printStackTrace();
+      throw new RuntimeException("Unexpected error", t);
+    }
+
+    if (classInfo.shouldInclude) {
+      if (Chicory.dump) {
+        try {
+          debug_transform.log(
+              "Dumping .class and .javap for %s to %s%n", binaryClassName, debug_bin_dir);
+          // Define the output file
+          Path outputFile = Paths.get(debug_bin_dir.toString(), binaryClassName + ".class");
+          // Write the byte array to a .class file.
+          Files.write(outputFile, newBytes);
+          // Write a BCEL like file with an extension of .javap.
+          ClassModel newClassModel = classFile.parse(newBytes);
+          try (BufferedWriter writer =
+              Files.newBufferedWriter(
+                  new File(debug_bin_dir, binaryClassName + ".javap").toPath(),
+                  StandardCharsets.UTF_8)) {
+            writer.write(newClassModel.toDebugString());
+          }
+        } catch (Throwable t) {
+          System.err.printf(
+              "Unexpected error %s dumping out debug files for: %s%n", t, binaryClassName);
+          t.printStackTrace();
+          // proceed with instrumentation
+        }
+      }
+      return newBytes;
+    } else {
+      debug_transform.log("Didn't instrument %s%n", binaryClassName);
+      // No changes to the bytecodes
+      return null;
+    }
+  }
+
+  /**
+   * Instrument the current class.
```

Comment:
Should this be "the given class"?  What is "the current class"?

Reply:
I don't see the difference.  All of Instrument operates on a single class at a time via the ClassfileTransformer interface.